### PR TITLE
Add action workflow to check code and build firmware automaticlly

### PR DESCRIPTION
**Pull request by DusKing1**
Use github action to build firmware.

Firmware-artifact contains all generated files.

AM32-F051-${RunNumber} contains only hex files, for blheli-configurator

### CODE REVIEW

**Comment by DusKing1** [comment]
Test passed on our fork:
https://github.com/ColinNiu/AM32-MultiRotor-ESC-firmware/actions/runs/465412368

@mikeller Do you need any change so for to get blheli-configurator can use the automatically generated hex files for flashing

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      - name: Get Pull Request Number
+        id: get_pullno
+        run: echo "PULL_NUMBER=$(echo "$GITHUB_REF" | awk -F / '{print $3}')" >> $GITHUB_ENV
+        if: startsWith(github.ref, 'refs/pull/')
+
+      - name: Make artifact name
+        id: make_artifactname
+        run: |
+          ARTIFACT_NAME="AM32-F051-${{ github.run_number }}"
```
Comment (mikeller) [inline .github/workflows/main.yml:40]: I like the naming - we should add the `AM32-F051-` prefix to the makefile.

**Review by mikeller — APPROVED** [review]

**Comment by mikeller** [comment]
@DusKing1: Looks ok to me. Currently configurator is only designed to load hex files that are explicitly listed in the manifest, so it will not be able to load development builds. But we can look into adding this.

File: .github/workflows/main.yml (hunk 1 above)
Comment (DusKing1) [reply .github/workflows/main.yml:40]: well, TBH this is the part I'm not sure about. If in this repo in future there will be support for F3 chips, the F051 name might causing misunderstanding. In general, I agree to change prefix in makefile.

**Comment by DusKing1** [comment]
@mikeller I don't know if configurator can load zip file. Those hex files are packed into zip and then uploaded

**Comment by DusKing1** [comment]
Made some minor changes with suggestions from @ColinNiu. Test result is here: https://github.com/ColinNiu/AM32-MultiRotor-ESC-firmware/actions/runs/473176858

**Review by mikeller — APPROVED** [review]

**Pull request merged**